app/models/employee.rb

```diff
@@ -9,8 +9,10 @@ class Employee < ApplicationRecord
 
   has_many :given_kudos, class_name: 'Kudo', foreign_key: 'giver_id', inverse_of: 'giver', dependent: :destroy
   has_many :received_kudos, class_name: 'Kudo', foreign_key: 'reciever_id', inverse_of: 'reciever', dependent: :destroy
+  has_many :orders, dependent: :destroy
```

Comment:
It is always a moot point for me.
Do you want to delete orders for a product when it is no longer available?

Reply:
No really. I could go with :nullify
However what will happen when I delete product that is referenced by Order: 
1. Will DB complain about it, schema has not_null on reward_id constraint set. 
2. When we list orders. I will have empty reward reference and views (probably controllers too) will blow up. I could ensure that reward is always set with empty stub, is this better approach?

Reply:
You can do `:nullify` but you need to set reward_id in DB to permit this behaviour otherwise, there will be an error.